task-13/src/main/java/com/epam/winter_java_lab/task_13/repos/MessageRepositoryCustomImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.epam.winter_java_lab.task_13.repos;
+
+import com.epam.winter_java_lab.task_13.domain.Message;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.util.List;
+
+@Repository
+public class MessageRepositoryCustomImpl implements MessageRepositoryCustom{
+
+    private EntityManager em;
+
+    public MessageRepositoryCustomImpl(EntityManager em) {
+        this.em = em;
+    }
+
+    @Override
+    public List<Message> search(String text, String tag) {
```

Comment:
Why do we need a custom implementation of this method?
Could we use spring data repository?